boards/arm/imxrt/imxrt1064-evk/src/imxrt_spi.c

```diff
@@ -0,0 +1,213 @@
+/****************************************************************************
+ * boards/arm/imxrt/imxrt1064-evk/src/imxrt_spi.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <errno.h>
+#include <debug.h>
+
+#include <nuttx/spi/spi.h>
+#include <arch/board/board.h>
+
+#include "arm_arch.h"
+
+#include "imxrt_config.h"
+#include "imxrt_lpspi.h"
+#include "imxrt_gpio.h"
+#include "imxrt1064-evk.h"
+
+#if defined(CONFIG_IMXRT_LPSPI1) || defined(CONFIG_IMXRT_LPSPI2) || \
+    defined(CONFIG_IMXRT_LPSPI3) || defined(CONFIG_IMXRT_LPSPI4)
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: imxrt_spidev_initialize
+ *
+ * Description:
+ *   Called to configure SPI chip select GPIO pins for the imxrt1060-evk
+ *   board.
+ *
+ ****************************************************************************/
+
+void weak_function imxrt_spidev_initialize(void)
+{
+#ifdef CONFIG_IMXRT_LPSPI1
+  imxrt_config_gpio(GPIO_LPSPI1_CS); /* LPSPI1 chip select */
+  imxrt_config_gpio(GPIO_MMCSD_EN);
```

Comment:
Looking at the schematic - the board supports full SDMMC is this for an external use case or a copy/paste artifact?

![image](https://user-images.githubusercontent.com/1945821/113858751-f3e94780-9758-11eb-851d-c79cff9af376.png)

Reply:
I believe a copy/paste artifact because I have took a MIMXRT1060-EVK as basis.

Reply:
I think it would make sense to have the board support only support what is on the schematic. Given the obscurity of IO definitions, it is far from obvious what IO pin GPIO_MMCSD_EN will be changes if SPI1 is lit.   So it is best to save some one the heartache.

Reply:
Agree with you. Will look into.

As you can see some build checks fail because of .ARM.exidx section and bigger offset then 0x3fffffff when using flash-ocram.ld linker script. This is not the case with MIMXRT1060_EVK because it is using FLASH_BASE of 0x60000000, while MIMXRT1064-EVK is using FLASH_BASE of 0x70000000 and this is out of R_ARM_PREL31 range. I do not know what to do here.

Reply:
First find out what is being emitted to that section.  Are exception (check command line with `V=1 make` in local build) and unwinding disabled (check the config.h in a local build)?